index.js

```diff
@@ -70,81 +70,75 @@ if (!output && !dir) {
   throw new Error('Must pass --output, --dir, or --replace option')
 }
 
-console.log(chalk.bold.red(logo))
+console.warn(chalk.bold.red(logo)) // Use warn to avoid writing to stdout
 
-spinner.text = `Loading Files`
+spinner.text = `Loading Config`
 spinner.start()
-globber(input).then((files) => {
-  if (files && files.length) spinner.succeed()
-  else throw new Error('You must pass a list of files to parse')
+Promise.all([globber(input), postcssrc()]).then((arr) => {
+  // Until parameter destructuring is supported:
+  let files = arr[0]
+  let config = arr[1]
 
-  spinner.text = `Loading Config`
-  spinner.start()
-  return postcssrc().then((config) => {
-    return Promise.all(files.map(file => {
-      return fs.readFile(file)
-        .then(css => {
-          config !== undefined ? spinner.succeed() : spinner.fail()
-
-          spinner.text = `Processing ${file}`
-          spinner.start()
-
-          options = Object.assign(
-            options,
-            { from: file, to: output || path.join(dir, path.basename(file)) }
-          )
-
-          return postcss(config.plugins)
-         .process(css, Object.assign(options, config.options))
-        })
-       .then((result) => {
-         if (path.extname(options.to) !== '.css') {
-           options.to = options.to.replace(/.\w+$/, '.css')
-         }
-
-         if (result.messages.some(i => i.type === 'warning')) spinner.fail()
-
-         return fs.outputFile(options.to, result.css)
-       })
-       .then(() => {
-         spinner.succeed()
-       })
-    }))
-  })
+  if (!files || !files.length) throw new Error('You must pass a list of files to parse')
+
+  spinner.succeed()
+
+  return Promise.all(files.map(file => process(file, config)))
+})
+.then(function () {
+  if (argv.watch) {
+    spinner.text = 'Waiting for file changes...'
+
+    watcher
+    .watch(input)
+    .on('ready', (file) => spinner.start())
+    .on('change', (file) => {
+      spinner.text = `Processing ${chalk.green(`${file}`)}`
+
+      postcssrc().then((config) => {
+        return process(file, config, watcher)
+      })
+      .then(() => {
+        spinner.text = 'Waiting for file changes...'
+        spinner.start()
+      })
+      .catch(errorHandler)
+    })
+  }
 })
 .catch(errorHandler)
 
-if (argv.watch) {
-  spinner.text = 'Waiting for file changes...'
-
-  watcher
-  .watch(input)
-  .on('ready', (file) => spinner.start())
-  .on('change', (file) => {
-    spinner.text = `Processing ${chalk.green(`${file}`)}`
+function process(file, config, watcher) {
+  spinner.text = `Processing ${file}`
+  spinner.start()
 
-    postcssrc().then((config) => {
-      return postcss(config.plugins)
-      .process(fs.readFileSync(file), config.options)
-    })
-    .then((result) => {
+  let options = Object.assign(
+    {},
+    defaultOptions,
+    config.options,
+    { from: file, to: output || path.join(dir, path.basename(file)) }
```

Comment:
Maybe `path.resolve` indead of `path.join` which gets normalize automatically for better/ensured Windows support.